api/webhooks/webhooks.py

```diff
@@ -43,97 +49,205 @@ class WebhookType(enum.Enum):
 }
 
 
-def get_webhook_model(webhook_type: WebhookType) -> typing.Union[WebhookModels]:
+def get_webhook_model(
+    webhook_type: WebhookType,
+) -> Type[Union[OrganisationWebhook, Webhook]]:
     if webhook_type == WebhookType.ORGANISATION:
         return OrganisationWebhook
     if webhook_type == WebhookType.ENVIRONMENT:
         return Webhook
 
 
-def call_environment_webhooks(environment, data, event_type):
+@register_task_handler()
+def call_environment_webhooks(
+    environment_id: int,
+    data: typing.Mapping,
+    event_type: str,
+    retries: int = WEBHOOK_BACKOFF_RETRIES,
+):
+    """
+    Call environment webhooks.
+
+    :param environment_id: The ID of the environment for which webhooks will be triggered.
+    :param data: A mapping containing the data to be sent in the webhook request.
+    :param event_type: The type of event to trigger the webhook.
+    :param retries: The number of times to retry the webhook in case of failure (int, default is 3).
+    """
+
     if settings.DISABLE_WEBHOOKS:
         return
-
+    try:
+        environment = Environment.objects.get(id=environment_id)
+    except Environment.DoesNotExist:
+        return
     _call_webhooks(
         environment.webhooks.filter(enabled=True),
         data,
         event_type,
         WebhookType.ENVIRONMENT,
+        retries,
     )
 
 
-def call_organisation_webhooks(organisation, data, event_type):
+@register_task_handler()
+def call_organisation_webhooks(
+    organisation_id: int,
+    data: typing.Mapping,
+    event_type: str,
+    retries: int = WEBHOOK_BACKOFF_RETRIES,
+):
+    """
+    Call organisation webhooks.
+
+    :param organisationt_id: The ID of the organisationt for which webhooks will be triggered.
+    :param data: A mapping containing the data to be sent in the webhook request.
+    :param event_type: The type of event to trigger the webhook.
+    :param retries: The number of times to retry the webhook in case of failure (int, default is 3).
+    """
+
     if settings.DISABLE_WEBHOOKS:
         return
-
+    try:
+        organisation = Organisation.objects.get(id=organisation_id)
+    except Organisation.DoesNotExist:
+        return
     _call_webhooks(
         organisation.webhooks.filter(enabled=True),
         data,
         event_type,
         WebhookType.ORGANISATION,
+        retries,
     )
 
 
 def call_integration_webhook(config, data):
```

Comment:
2 points here: 

 1. While we're here, can we add typing to this function too? 
 2. Is there a reason that we're not implementing the back off logic to this webhook implementation?

Reply:
1. Sure, I will add it. 
2. I had second thoughts on synchronous backoffs on the main thread. If that's fine, I can add in backoff for 3 retries and max backoff time to `2s` maybe?

Reply:
Regarding point (2) here it's actually not run in the main thread, it's run in a daemon thread dispatched to by the main thread (and never checked afterwards). The logic here is quite convoluted and I'm not exactly happy with it, but it's served us well up to this point. 

If you follow the usages of this `call_integration_webhook`, you'll find that it's called from `integrations.webhook.WebhookWrapper` which inherits from `integrations.common.wrapper.AbstractBaseIdentityIntegrationWrapper` that has the `identify_user_async` method exposed (and is the only method actually called by the main thread). 

I'd love to refactor all of this to use task handlers instead of these daemon threads directly (the initial refactor would likely just be to use `run_in_thread`) but that's for another PR.  

Maybe _all_ of this should be handled in a separate PR to be honest and we'll leave the integrations webhook to only try once for now. I'll leave it up to you here.

Reply:
I have added backoff to it, using the `backoff` library. I can't use a task processor here, because the return values of `_call_webhooks` are needed in calling functions.